EID-312 Enable European Users

lamerexter opened the pull request:
- This is a re-application of EID-293 to reduce complexity of the MatchingServiceResource via a simple delegation model.
- HealthCheck, Verify AttributeQuery handling and Eidas AttributeQuery handling are all given their own single-purpose service (think 12-factor...)
- Result is that MatchingServiceResource::receiveSoapRequest method is reduced to 3 lines...
- All tests run green, as expected.

lamerexter commented on src/main/java/uk/gov/ida/matchingserviceadapter/MatchingServiceAdapterModule.java:
```diff
@@ -130,6 +149,88 @@ protected void configure() {
+            new EidasMatchingService(
+                new EidasAttributeQueryValidator(
+                    verifyMetadataResolver,
+                    cmr,
```
This is rather short for countryMetadataResolver

thomasalanlee commented:
This is too large. Perhaps introducing the ServiceLocator could be a separate change

The commit messages are not descriptive enough. If you are introducing a significant change (e.g. the ServiceLocator) please describe why in the commit messages.

There are quite a few places acronyms or abbreviations have been used. We should favour clarity over brevity (e.g. ctor, cmr)

lamerexter commented on src/test/java/uk/gov/ida/matchingserviceadapter/services/EidasMatchingServiceTest.java:
```diff
@@ -0,0 +1,67 @@
+package uk.gov.ida.matchingserviceadapter.services;
+
+import org.beanplanet.messages.domain.Message;
```
Don't worry: all references to the Beanplanet library HAVE BEEN REMOVED in the next commit changeset!

adityapahuja commented:
Test this please.

lamerexter commented:
Thanks @thomasalanlee.  I have addressed all of your review items:

1. - Beanplanet library references - all were already removed by the next commit in the list.
2. - PR Job build failure - Jenkins is being restarted
3. - Commit messages - messages made more descriptive.
4. - Acronyms - ctor and cmr renamed accordingly.

Thanks.

adityapahuja commented:
Test this please.

adityapahuja commented:
Test this please.

TimWSpence commented:
test this please

TimWSpence commented:
test this please

adityapahuja commented:
Test this please.

lamerexter commented:
Test this please.

thomasalanlee commented on src/test/java/uk/gov/ida/matchingserviceadapter/validators/datetime/JodaAndJavaDateTimeConvertersTest.java:
```diff
@@ -16,7 +16,7 @@
 public class JodaAndJavaDateTimeConvertersTest {
     @Test
-    public void ctorForTestCoverageOnly() {
+    public void constructorForTestCoverageOnly() {
```
I'm not sure we need this test do we?

thomasalanlee commented on src/test/java/uk/gov/ida/matchingserviceadapter/domain/VerifyMatchingServiceResponseTest.java:
```diff
@@ -9,7 +9,7 @@
 public class VerifyMatchingServiceResponseTest {
     @Test
-    public void ctor() {
+    public void constructor() {
```
Please describe what behaviour the test is verifying in the test title.

thomasalanlee commented on src/test/java/uk/gov/ida/matchingserviceadapter/resources/DelegatingMatchingServiceResponseRendererTest.java:
```diff
@@ -19,7 +19,7 @@
 
     @Test
-    public void ctor() {
+    public void constructor() {
```
Please describe the behaviour we're verifying in the test name

thomasalanlee commented on src/test/java/uk/gov/ida/matchingserviceadapter/services/EidasMatchingServiceTest.java:
```diff
@@ -26,7 +26,7 @@
     @SuppressWarnings("unchecked")
     @Test
-    public void ctor() {
+    public void constructor() {
```
Please describe the behaviour we're verifying in the test name

thomasalanlee commented on src/test/java/uk/gov/ida/matchingserviceadapter/validators/IssueInstantValidatorTest.java:
```diff
@@ -29,7 +29,7 @@
 
     @Test
-    public void ctorTtlFutureMessageDateTimeProviderAndDurations() {
+    public void constructorWithTtlFutureMessageDateTimeProviderAndDurations() {
```
Please describe the behaviour we're verifying in the test name

thomasalanlee commented on src/test/java/uk/gov/ida/matchingserviceadapter/validators/datetime/DurationNotExceededValidatorTest.java:
```diff
@@ -23,7 +23,7 @@
 
     @Test
-    public void ctorMessageFromToAndMaximum() {
+    public void constructorMessageFromToAndMaximum() {
```
Please describe the behaviour we're verifying in the test name

thomasalanlee commented on src/test/java/uk/gov/ida/matchingserviceadapter/validators/datetime/DurationNotExceededValidatorTest.java:
```diff
@@ -41,7 +41,7 @@ public void ctorMessageFromToAndMaximum() {
 
     @Test
-    public void ctorMessageFromAndMaximum() {
+    public void constructorMessageFromAndMaximum() {
```
Please describe the behaviour we're verifying in the test name

thomasalanlee commented on src/test/java/uk/gov/ida/matchingserviceadapter/validators/datetime/NotInTheFutureValidatorTest.java:
```diff
@@ -21,7 +21,7 @@
 
     @Test
-    public void ctorMessageAndAtProvider() {
+    public void constructorMessageAndAtProvider() {
```
Please describe the behaviour we're verifying in the test name

thomasalanlee commented on src/main/java/uk/gov/ida/matchingserviceadapter/services/DelegatingMatchingService.java:
```diff
@@ -31,13 +31,18 @@ public DelegatingMatchingService(ServiceLocator<MatchingServiceRequestContext, M
-        return matchingServiceLocator.findServiceFor(requestContext).handle(requestContext);
+
+        MatchingService delegateService = matchingServiceLocator.findServiceFor(requestContext);
+        if (delegateService == null) {
```
We should probably use Optional.orElseThrow() to handle null checks

thomasalanlee commented on src/test/java/uk/gov/ida/matchingserviceadapter/domain/VerifyMatchingServiceResponseTest.java:
```diff
@@ -0,0 +1,19 @@
+
+public class VerifyMatchingServiceResponseTest {
+    @Test
+    public void ctor() {
```
Please rename this test to describe what behaviour it is verifying (or delete if we don't need it)

thomasalanlee commented on src/test/java/uk/gov/ida/matchingserviceadapter/resources/MatchingServiceResourceTest.java:
```diff
@@ -24,7 +28,9 @@
 import java.util.jar.Attributes;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.hamcrest.CoreMatchers.sameInstance;
```
Please switch to use AssertJ
